### PR TITLE
fix: savepoints must be rolled back and released

### PR DESCRIPTION
**Pull request by kantai**

### CODE REVIEW

File: src/vm/database/sqlite.rs [hunk 1]
```diff
@@ -154,3 +154,15 @@ impl SqliteConnection {
...
+
+#[cfg(test)]
+#[test]
+#[should_panic]
+fn test_rollback() {
+    let mut conn = SqliteConnection::memory().unwrap();
+    let bhh = BlockHeaderHash([1; 32]);
+    conn.begin(&bhh);
+    conn.rollback(&bhh);
+    conn.commit(&bhh); // shouldn't be on the stack!
```
Comment (lgalabru) [inline src/vm/database/sqlite.rs:167]: Should we write this test to make clear that the panic should happen on `commit`?

**Comment by jcnelson** [comment]
I've added a test in my relayer branch that only passes with this enabled.  The relayer tests need to be able to generate and mine invalid blocks, and that's what lead to me discovering this problem.  Thanks for fixing it so quickly @kantai!

File: src/vm/database/sqlite.rs (hunk 1 above)
Comment (kantai) [reply src/vm/database/sqlite.rs:167]: Sure -- 70b62da54 makes this a more specific test.

Comment (lgalabru) [reply src/vm/database/sqlite.rs:167]: Thank you @kantai!

**Pull request merged**